presto-main/src/main/java/io/prestosql/sql/planner/DynamicFilterSupplier.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.predicate.TupleDomain;
+
+public interface DynamicFilterSupplier
+{
+    DynamicFilterSupplier EMPTY = new DynamicFilterSupplier() {
+        @Override
+        public ListenableFuture<?> getBlockingFuture()
+        {
+            return Futures.immediateFuture(null);  // no blocking
+        }
+
+        @Override
+        public TupleDomain<ColumnHandle> getCurrentPredicate()
+        {
+            return TupleDomain.all();  // no filtering
+        }
+    };
+
+    ListenableFuture<?> getBlockingFuture();
```

Comment:
Call this `isBlocked`, similar to how we do it elsewhere (e.g., `ConnectorPageSource`)

Reply:
What are the semantics of this method with respect to `getCurrentPredicate`? Is it intended to block until there is *some* predicate available, even if the predicate continues to change after?

Reply:
> What are the semantics of this method with respect to `getCurrentPredicate`?

It should block until all dynamic filters are collected, or the specified timeout expires.
If the timeout expires, the predicate returned by `getCurrentPredicate` may change after all filters are eventually collected.

I have added the comment here, and also at `LocalDynamicFiltersCollector` and `LocalExecutionPlanner`.